client/src/components/Sales/Modal.jsx

```diff
@@ -0,0 +1,136 @@
+import useEmailValidation from 'hooks/useEmailValidation.jsx';
+import useSales from 'hooks/useSales.jsx';
+import PropTypes from 'prop-types';
+import { useQuery } from 'react-query';
+import { getBikes } from 'utils/api/bikes.js';
+import AboutBike from './AboutBike.jsx';
+
+const SalesModal = ({ showButton = false }) => {
+  const { isSuccess, data } = useQuery('bikes', getBikes);
+  const {
+    handleSubmit,
+    isSaleModalOpen,
+    onSaleModalOpen,
+    onSaleModalClose,
+    emailRef,
+    setSelectedBikeId,
+    setName,
+    setQuantity,
+    selectedBikeId,
+    isLoadingButton,
+    name,
+  } = useSales();
+  const { isValidEmail, handleEmailInput } = useEmailValidation();
+
+  return (
+    <>
+      {showButton ? (
+        <Center mt={4}>
+          <Button onClick={onSaleModalOpen}>Add Sale</Button>
+        </Center>
+      ) : (
+        <IconButton
+          colorScheme="blue"
+          variant="outline"
+          aria-label="add"
+          float="right"
+          m={2}
+          icon={<SmallAddIcon />}
+          onClick={onSaleModalOpen}
+        />
+      )}
+      <Modal isOpen={isSaleModalOpen} onClose={onSaleModalClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Add a sale</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <FormControl isRequired>
+              <FormLabel>Bike</FormLabel>
+              <Select onChange={(e) => setSelectedBikeId(e.target.value)} defaultValue="default">
+                <option disabled value="default">
+                  Select
+                </option>
+                {isSuccess &&
+                  data.data
+                    .filter((bike) => bike.stock > 0)
+                    .map((bike) => (
+                      <option key={bike._id} value={bike._id}>
+                        {bike.name}
+                      </option>
+                    ))}
+              </Select>
+            </FormControl>
+            <FormControl isRequired mt={4}>
+              <FormLabel>Customer Email</FormLabel>
+              <Input
+                placeholder="Customer Email"
+                type="email"
+                ref={emailRef}
+                focusBorderColor={isValidEmail ? 'blue.500' : 'red'}
+                isInvalid={!isValidEmail}
+                onChange={handleEmailInput}
+                required
+              />
+            </FormControl>
+            <FormControl isRequired mt={4}>
+              <FormLabel>Customer Name</FormLabel>
+              <Input onChange={(e) => setName(e.target.value)} placeholder="Customer Name" />
+            </FormControl>
+            <FormControl isRequired mt={4}>
+              <FormLabel>Quantity</FormLabel>
+              <NumberInput min={1} defaultValue={1}>
+                <NumberInputField onChange={(e) => setQuantity(e.target.value)} />
+                <NumberInputStepper>
+                  <NumberIncrementStepper />
```

Comment:
I think you need to call the method setQuantity here since it wouldn't record the input quantity. This leads to having an order quantity of 1

Reply:
I switched to using a Ref (I can't even remember why I used a `useState` in the first place for this.